cupy/cuda/cub.pyx

```diff
@@ -91,10 +91,10 @@ cpdef _preprocess_array(ndarray arr, axis, bint keepdims, str order):
     return out_shape, contiguous_size
 
 
-def device_reduce(ndarray x, op, out=None, bint keepdims=False):
+def device_reduce(ndarray x, int op, out=None, bint keepdims=False):
     cdef ndarray y, z
     cdef memory.MemoryPointer ws
-    cdef int dtype_id, ndim_out, kv_bytes
+    cdef int dtype_id, ndim_out, kv_bytes, x_size
```

Comment:
Shouldn't this also be `size_t`? Proposing changing the type in PR ( https://github.com/cupy/cupy/pull/3117 ).